### ISSUE TITLE
False Positive: S2933

### ISSUE THREAD

**Issue by aPlacuzzi** [SonarSource/sonar-dotnet#4858]
### Description

Tuple deconstruction breaks S2933 (make field readonly)

### Repro steps

```c#
class Program
{
    private int x; // S2933: make 'x' 'readonly'
    private int y; // S2933: make 'y' 'readonly'

    public Program()
    {
        x = 0;
        y = 0;
    }

    public void ModifyXAndY()
    {
        (x, y) = NewValues();
    }

    private (int, int) NewValues() => (1, 2);
}
```

### Expected behavior

I don't expect any warning.

### Actual behavior

I get S2933 in `x` and `y` definition.
If I define `x` and `y` as `readonly` fields, I get compilation error: 	"A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer)"

### Known workarounds

No.

### Related information

**C#**
Version: 7.3
.NetFramework 4.72

**Visual Studio version**
16.10.4

**MSBuild / dotnet version**
```
msbuild -ver
Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

16.10.2.30804
```

**Sonar Nuget Package**
version: 8.27.0.35380

**Operating System**
OS Name: Microsoft Windows 10 Pro
Version: 10.0.19043 Build 19043

**Comment by csaba-sagi-sonarsource** [comment]
Hi @aPlacuzzi, thank you for reporting the issue. I can confirm it as a false positive.
